aws-lc-rs/src/cipher.rs

```diff
@@ -873,6 +941,82 @@ fn decrypt_aes_cfb_mode<'in_out>(
     Ok(in_out)
 }
 
+#[allow(clippy::needless_pass_by_value, clippy::unnecessary_wraps)]
+fn encrypt_aes_ecb_mode(
+    key: &SymmetricCipherKey,
+    context: EncryptionContext,
+    in_out: &mut [u8],
+) -> Result<DecryptionContext, Unspecified> {
+    if !matches!(context, EncryptionContext::None) {
+        unreachable!();
+    }
+
+    #[allow(clippy::match_wildcard_for_single_variants)]
+    let key = match &key {
+        SymmetricCipherKey::Aes128 { enc_key, .. } | SymmetricCipherKey::Aes256 { enc_key, .. } => {
+            enc_key
+        }
+        _ => unreachable!(),
+    };
+
+    let mut in_out_iter = in_out.chunks_exact_mut(AES_BLOCK_LEN);
+
+    for block in in_out_iter.by_ref() {
+        aes_ecb_encrypt(key, block);
+    }
+
+    // This is a sanity check that should not happen. We validate in `encrypt` that in_out.len() % block_len == 0
+    // for this mode.
+    assert!(in_out_iter.into_remainder().is_empty());
+
+    Ok(context.into())
+}
+
+#[allow(clippy::needless_pass_by_value, clippy::unnecessary_wraps)]
+fn decrypt_aes_ecb_mode<'in_out>(
+    key: &SymmetricCipherKey,
+    context: DecryptionContext,
+    in_out: &'in_out mut [u8],
+) -> Result<&'in_out mut [u8], Unspecified> {
+    if !matches!(context, DecryptionContext::None) {
+        unreachable!();
+    }
+
+    #[allow(clippy::match_wildcard_for_single_variants)]
+    let key = match &key {
+        SymmetricCipherKey::Aes128 { dec_key, .. } | SymmetricCipherKey::Aes256 { dec_key, .. } => {
+            dec_key
+        }
+        _ => unreachable!(),
+    };
+
+    {
+        let mut in_out_iter = in_out.chunks_exact_mut(AES_BLOCK_LEN);
+
+        for block in in_out_iter.by_ref() {
+            aes_ecb_decrypt(key, block);
+        }
+
+        // This is a sanity check hat should not fail. We validate in `decrypt` that in_out.len() % block_len == 0 for
+        // this mode.
+        assert!(in_out_iter.into_remainder().is_empty());
+    }
+
+    Ok(in_out)
+}
+
+fn aes_ecb_encrypt(key: &AES_KEY, in_out: &mut [u8]) {
+    indicator_check!(unsafe {
+        AES_ecb_encrypt(in_out.as_ptr(), in_out.as_mut_ptr(), key, AES_ENCRYPT);
+    });
+}
+
+fn aes_ecb_decrypt(key: &AES_KEY, in_out: &mut [u8]) {
+    indicator_check!(unsafe {
+        AES_ecb_encrypt(in_out.as_ptr(), in_out.as_mut_ptr(), key, AES_DECRYPT);
+    });
+}
```

Comment:
Should we migrate calls to [`cipher::aes::encrypt_block_aes`](https://github.com/aws/aws-lc-rs/blob/c2612bf996810d0397b52b6bf706815f36e83332/aws-lc-rs/src/cipher/aes.rs#L25) to call here instead?